### DISCUSSION TITLE
Retrieve Provider from SignInScreen actions

### DISCUSSION THREAD

**Discussion by jeanmatthieud** [firebase/FirebaseUI-Flutter#74] [Q&A] [answered]
Hi,

I use a SignInScreen, and I want to be able to retrieve the current provider during `AuthStateChangeAction`.

eg.:
```dart
SignInScreen(
  // Providers are set with FirebaseUIAuth.configureProviders()
  // providers: [],
  actions: [
    AuthStateChangeAction<AuthFailed>((context, state) {
      // I want to be able to manage the AuthFailed state according to the provider
      // or even call a provider function to recover from the error
    }),
  ],
)
```

How can we know which provider is linked to the current `AuthStateChangeAction` callback?

**Comment by lesnitsky** [comment, marked as answer]
There's currently no way to determine which provider was used in this callback and it's a reasonable ask, so I will be looking into implementing this in future releases. Tracked under #75
